### DISCUSSION TITLE
Optimizing budget on subset of paid media

### DISCUSSION THREAD

**Discussion by cjyoung-picpay** [facebookexperimental/Robyn#378] [Ideas]
First off, thanks for the implementation. Really good work all around. 

Are there any plans to include different budget optimization features? I'm looking at freezing some paid media budgets (because for example they are outside of my control), while optimizing the remaining ones. 

Since the the code and documents are well organized, I can see that the budget optimizer uses nlop with AUGLAG + SLSQP, so I can probably recreate the budget optimization on my end. Food for thought all the same.

**Comment by kyletgoldberg** [comment]
Hi - you should be able to freeze some individual channel budgets by setting the upper and lower constraints for a given channel to 1 and it will essentially just optimize the remaining ones. Try that out and let us know if that covers your use case

**Reply by cjyoung-picpay** [reply]
Appreciated. I'll get back to you on this.
